src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.ts

```diff
@@ -0,0 +1,388 @@
+        return this.currentPageComponents.find((submissionComponent) => (submissionComponent as ExamSubmissionComponent).getExercise().id === this.activeExamPage.exercise?.id);
+    }
+
+    /**
+     * This method is called when the user clicks on the slider
+     * @param changeContext the change context of the slider
+     */
+    onInputChange(changeContext: ChangeContext) {
+        this.selectedTimestamp = changeContext.value;
+        const submission = this.findCorrespondingSubmissionForTimestamp(changeContext.value);
+        if (this.isSubmissionVersion(submission)) {
+            const submissionVersion = submission as SubmissionVersion;
+            this.currentExercise = submissionVersion.submission.participation?.exercise;
+        } else if (this.isFileUploadSubmission(submission)) {
+            const fileUploadSubmission = submission as FileUploadSubmission;
+            this.currentExercise = fileUploadSubmission.participation?.exercise;
+        } else {
+            const programmingSubmission = submission as ProgrammingSubmission;
+            this.currentExercise = programmingSubmission.participation?.exercise;
+        }
+        const exerciseIndex = this.studentExam.exercises!.findIndex((examExercise) => examExercise.id === this.currentExercise?.id);
+        this.exerciseIndex = exerciseIndex;
+        this.currentSubmission = submission;
+        this.examNavigationBarComponent.changePage(false, exerciseIndex, false, submission);
+    }
+
+    /**
+     * Finds the submission for the current timestamp.
+     * @param timestamp The timestamp for which the submission should be found.
+     */
+    private findCorrespondingSubmissionForTimestamp(timestamp: number): SubmissionVersion | ProgrammingSubmission | FileUploadSubmission | undefined {
+        const comparisonObject = dayjs(timestamp);
+        for (let i = 0; i < this.submissionVersions.length; i++) {
+            const submissionVersion = this.submissionVersions[i];
+            if (submissionVersion.createdDate.isSame(comparisonObject)) {
+                return submissionVersion;
+            }
+        }
+        for (let i = 0; i < this.programmingSubmissions.length; i++) {
+            const programmingSubmission = this.programmingSubmissions[i];
+            if (programmingSubmission.submissionDate?.isSame(comparisonObject)) {
+                return programmingSubmission;
+            }
+        }
+        for (let i = 0; i < this.fileUploadSubmissions.length; i++) {
+            const fileUploadSubmission = this.fileUploadSubmissions[i];
+            if (fileUploadSubmission.submissionDate?.isSame(comparisonObject)) {
+                return fileUploadSubmission;
+            }
+        }
+        return undefined;
+    }
+
+    /**
+     * helper method to check if the object is a file upload submission
```

Comment:
An explanation about why a simple `instanceof FileUploadSubmission` does not suffice would help here I think